contracts/token/psp1155/psp1155.rs

```diff
@@ -73,12 +73,15 @@ impl<T: PSP1155Storage + Flush> PSP1155 for T {
         self._transfer_guard(operator, from, to)?;
 
         let ids_amounts = vec![(id, amount)];
-        self._before_token_transfer(&from, &to, &ids_amounts)?;
-        self._do_safe_transfer_check(operator, from, to, ids_amounts, data)?;
 
-        self._transfer_from(from, to, id, amount)?;
+        self._do_safe_transfer_check(operator, from, to, ids_amounts.clone(), data)?;
+
+        self._before_token_transfer(Some(&from), Some(&to), &ids_amounts)?;
```

Comment:
Why did we change the ordering of `_before_token_transfer` and `_do_safe_transfer_check`? I think local check should be before the cross-contract call.

Reply:
@sventime advised to do it as in OpenZeppelin